<a href="https://colab.research.google.com/github/Annette-1/SIMULACION-II/blob/main/Matrices_elevadas_CM.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# <span style="color:orange;"> Comportamiento a largo plazo de Cadenas de Markov</span>

Dada la matriz de probabiliddaes de transición $P$ y el estado inicial $\pi_0$ de una cadena de Markov, la distribución del tiempo siguiente se calculo como:
 $$\pi_0 =\pi_1 P
$$

In [ ]:
from sympy import *
init_printing(use_unicode=True)

In [ ]:
P=Matrix([[0.6,0.4],[0.2,0.8]])

In [ ]:
P

⎡0.6  0.4⎤
⎢        ⎥
⎣0.2  0.8⎦

Distribución inicial

$\pi_0= [1,0]$

In [ ]:
p=Matrix([[1,0]])

In [ ]:
p

[1  0]

Calculamos el vector para el tiempo $1$

In [ ]:
p0=p*P
p0

[0.6  0.4]

Probalidad del que el sábado llueva, lo cual corresponde a resolver la ecuación
$$
\pi_5= \pi_0 * P^5
$$

In [ ]:
p5= p*P**5

In [ ]:
p5

[0.34016  0.65984]

¿Cuál es la probabilidad de que a largo plazo un día sea soleado? Acompañe su respuesta de una gráfica.

**Cadenas de Markov: Ejemplo**

Tenemos la siguiente matrix de transición

$$P=
\begin{bmatrix}
0.80 & 0.15 & 0.05 \\
0.06 & 0.90 & 0.04 \\
0.04 & 0.06 & 0.90
\end{bmatrix}
$$